### PR TITLE
api: support getting blob content

### LINKED ISSUE #5979: Support API: `GET /repos/:owner/:repo/git/blobs/:file_sha`
Ref: https://developer.github.com/v3/git/blobs/#get-a-blob

### PR DESCRIPTION
**Pull request by jonatan5524**
### Describe the pull request

A clear and concise description of what the pull request is about, i.e. what problem should be fixed? 
Added a new API support for getting a blob

Link to the issue: fix #5979 

### Checklist

- [x] I agree to follow the [Code of Conduct](https://go.dev/conduct) by submitting this pull request.
- [x] I have read and acknowledge the [Contributing guide](https://github.com/gogs/gogs/blob/main/.github/CONTRIBUTING.md).
- [ ] I have added test cases to cover the new code.

### CODE REVIEW

**Comment by unknwon** [comment]
Thanks for the PR!

Could you provide some proof you've tested this new endpoint end-to-end?

**Comment by jonatan5524** [comment]
> Thanks for the PR!
> 
> Could you provide some proof you've tested this new endpoint end-to-end?

image

File: internal/route/api/v1/repo/blob.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	gitRepo, err := git.Open(repoutil.RepositoryPath(c.Params("username"), c.Params("reponame")))
+	if err != nil {
+		c.Error(err, "open repository")
+		return
+	}
+
+	sha := c.Params(":sha")
+	blob, err := gitRepo.CatFileBlob(sha)
+	if err != nil {
+		c.NotFoundOrError(gitutil.NewError(err), "get blob")
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:24]:
```suggestion
		c.NotFoundOrError(gitutil.NewError(err), "get blob")
		return
```

File: internal/route/api/v1/repo/blob.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		Content  string `json:"content"`
+		Encoding string `json:"encoding"`
+		URL      string `json:"url"`
+		Sha      string `json:"sha"`
+		Size     int64  `json:"size"`
+	}
+
+	content, err := blob.Blob().Bytes()
+	if err != nil {
+		c.NotFoundOrError(gitutil.NewError(err), "get blob content")
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:37]:
```suggestion
		c.NotFoundOrError(gitutil.NewError(err), "get blob content")
		return
```

File: internal/route/api/v1/repo/blob.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	c.JSONSuccess(&repoGitBlob{
+		Content:  base64.StdEncoding.EncodeToString(content),
+		Encoding: "base64",
+		URL:      fmt.Sprintf("%s/repos/%s/%s/git/blobs/%s", c.BaseURL, c.Params(":username"), c.Params(":reponame"), sha),
+		Sha:      sha,
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:44]:
```suggestion
		SHA:      sha,
```

**Review by unknwon — APPROVED** [review]
Thank you!

File: internal/route/api/v1/repo/blob.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/gogs/git-module"
+
+	"gogs.io/gogs/internal/context"
+	"gogs.io/gogs/internal/gitutil"
+	"gogs.io/gogs/internal/repoutil"
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:11]:
```suggestion
```

File: internal/route/api/v1/repo/blob.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	"fmt"
+
+	"github.com/gogs/git-module"
+
+	"gogs.io/gogs/internal/context"
+	"gogs.io/gogs/internal/gitutil"
+)
+
+func RepoGitBlob(c *context.APIContext) {
+	gitRepo, err := git.Open(c.Repo.Repository.RepoPath())
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:14]:
```suggestion
	gitRepo, err := git.Open(repoutil.RepositoryPath(c.Params(":username"), c.Params(":reponame")))
```

File: internal/route/api/v1/repo/blob.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package repo
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/gogs/git-module"
+
+	"gogs.io/gogs/internal/context"
+	"gogs.io/gogs/internal/gitutil"
```
Comment (unknwon) [inline internal/route/api/v1/repo/blob.go:10]:
```suggestion
	"gogs.io/gogs/internal/gitutil"
	"gogs.io/gogs/internal/repoutil"
```

**Review by unknwon — COMMENTED** [review]
My bad, used a deprecated method 🤦

**Comment by jonatan5524** [comment]
Thank you for the CR!

**Pull request merged**